### PR TITLE
fix profile creation bug; update/create profile with json

### PR DESCRIPTION
**Pull request by xlqian**
TMA-1881 
http://jira.canaltp.fr/browse/NAVITIAII-1881
- traveler_profile now can be updated by args and json
- fix bugs

### CODE REVIEW

File: source/tyr/tyr/resources.py [hunk 1]
```diff
@@ -570,31 +570,49 @@ def delete(self, id):
...
-        parser.add_argument('walking_speed', type=parser_args_type.float_gt_0, required=False, case_sensitive=False)
-        parser.add_argument('bike_speed', type=parser_args_type.float_gt_0, required=False, case_sensitive=False)
-        parser.add_argument('bss_speed', type=parser_args_type.float_gt_0, required=False, case_sensitive=False)
-        parser.add_argument('car_speed', type=parser_args_type.float_gt_0, required=False, case_sensitive=False)
-        parser.add_argument('wheelchair', type=parser_args_type.true_false, required=False, case_sensitive=False)
+        parser.add_argument('walking_speed', type=parser_args_type.float_gt_0, required=False, location=('json', 'args'))
+        parser.add_argument('bike_speed', type=parser_args_type.float_gt_0, required=False, location=('json', 'values'))
+        parser.add_argument('bss_speed', type=parser_args_type.float_gt_0, required=False, location=('json', 'values'))
+        parser.add_argument('car_speed', type=parser_args_type.float_gt_0, required=False, location=('json', 'values'))
+        parser.add_argument('wheelchair', type=parser_args_type.true_false, required=False, location=('json', 'values'))
```
Comment (TeXitoi) [inline source/tyr/tyr/resources.py]: long lines

File: source/tyr/tyr/resources.py [hunk 2]
```diff
@@ -657,10 +680,10 @@ def put(self, name=None, traveler_type=None):
         if profile is None:
             return {'error': 'Non profile is found to update'}, 404
         try:
-            for (attr, default_value) in default_traveler_profile_params[traveler_type].iteritems():
+            for (attr, args_value) in self.args.iteritems():
                 # override hardcoded values by args if args are not None
-                value = default_value if self.args.get(attr) is None else self.args.get(attr)
-                setattr(profile, attr, value)
+                if args_value is not None:
+                    setattr(profile, attr, args_value)
```
Comment (antoine-de) [inline source/tyr/tyr/resources.py:679]: not sure it's the same, in the previous version, the attribute was set no ?

File: source/tyr/tyr/resources.py [hunk 3]
```diff
@@ -570,31 +570,54 @@ def delete(self, id):
...
+                            help='in second', location=('json', 'values'))
         parser.add_argument('max_bss_duration_to_pt', type=parser_args_type.float_gt_0, required=False,
-                            case_sensitive=False)
+                            help='in second', location=('json', 'values'))
         parser.add_argument('max_car_duration_to_pt', type=parser_args_type.float_gt_0, required=False,
-                            case_sensitive=False)
+                            help='in second', location=('json', 'values'))
         parser.add_argument('first_section_mode[]',
-                            type=parser_args_type.option_value(['walking', 'car', 'bss', 'bike']),
-                            required=False, action='append', case_sensitive=False, dest='first_section_mode')
```
Comment (pbougue) [inline source/tyr/tyr/resources.py:594]: You can let the `case_sensitive` here and next line

File: source/tyr/tyr/resources.py (hunk 2 above)
Comment (xlqian) [inline source/tyr/tyr/resources.py:679]: it's no longer the same...  in "PUT" method, I **update** only attributes that are mentioned by user, on contrary, the old version update all attributes even though those are not  wanted by user... it's a bug...

File: source/tyr/tyr/resources.py (hunk 3 above)
Comment (xlqian) [inline source/tyr/tyr/resources.py:594]: done

File: source/navitiacommon/navitiacommon/parser_args_type.py [hunk 4]
```diff
@@ -47,20 +47,28 @@ def float_gt_0(value, name):
...
         raise ValueError("The {} argument must be true or false, you gave : {}"
                          .format(name, value))
 
 
-def option_value(values):
+def option_value(optional_values):
     def to_return(value, name):
-        if not (value in values):
+        # if input value is iterable
+        if hasattr(value, '__iter__'):
```
Comment (antoine-de) [inline source/navitiacommon/navitiacommon/parser_args_type.py:64]: :+1:  I find this way clearer than the previous version

Comment (xlqian) [inline source/navitiacommon/navitiacommon/parser_args_type.py:64]: :yellow_heart:

**Pull request merged**